webrender/src/frame_builder.rs

```diff
@@ -716,8 +716,13 @@ impl FrameBuilder {
         // The local space box shadow rect. It is the element rect
         // translated by the box shadow offset and inflated by the
         // box shadow spread.
-        let bs_rect = box_bounds.translate(box_offset)
-                                .inflate(spread_radius, spread_radius);
+        let mut bs_rect = box_bounds.translate(box_offset);
```

Comment:
This could probably be written something like:

```
let inflate_amount = match clip_mode { ... };
```

Which just negates the amount, and then keep the original code to calculate the bs_rect.